client/mapping-sync/src/lib.rs

```diff
@@ -112,12 +113,10 @@ where
 		if leaves.is_empty() {
 			return Ok(false);
 		}
-
 		current_syncing_tips.append(&mut leaves);
 	}
 
 	let mut operating_tip = None;
-
 	while let Some(checking_tip) = current_syncing_tips.pop() {
 		if !frontier_backend
```

Comment:
You can put the `operating_header.number() < &sync_from` logic as an additional cause here.

If the operating header is less than `sync_from`, it doesn't mean other tips syncing are also less than `sync_from` and we may still want to continue syncing them.

I think the parachain strategy is also flawed now but we can fix that later.

Reply:
This is a better way, Thanks. Please take a review again later.